packages/dds/tree/src/test/simple-tree/objectFactory.spec.ts

```diff
@@ -19,35 +20,37 @@ import { getRoot } from "./utils.js";
 describe("SharedTreeObject factories", () => {
 	const sb = new SchemaFactory("test");
 
-	class ChildA extends sb.object("childA", {
+	const ChildA = sb.object("childA", {
```

Comment:
Curious: are these pattern changes required? Do they represent the patterns we are encouraging for our customers? Just wondering what the implications of this are for end-users of the APIs and whether or not we should be testing both patterns here.

Reply:
It's the same tradeoffs between classes and object literals ({}) in JavaScript:

`class Foo extends _.object(..) {}` is a subclass that can have custom members, which is super convenient for domain modeling, but won't roundtrip through JSON, etc.

`const Foo = _.object(..);` is a plain data type that is indistinguishable from a '{}' literal, which makes it more interoperable with pre-existing code and external systems.

In this case, I converted to a plain data type to re-enable the tests below that use 'assert.deepEqual(..)'. (The classes fail deepEqual because their prototype chain is different than a '{}'.)